### PR TITLE
Change npm so that config get/set uses `any` instead of `string`

### PR DESCRIPTION
**Pull request by styfle**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [see usage](https://github.com/npm/cli/blob/59e5056a2129cb2951f4ff3b657ada20657f01a7/lib/install.js#L242) and [see defaults](https://github.com/npm/cli/blob/550bf703ae3e31ba6a300658ae95b6937f67b68f/lib/config/defaults.js#L112)
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Review by SimonSchick — CHANGES_REQUESTED** [review]

File: types/npm/index.d.ts [hunk 1]
```diff
@@ -167,8 +167,8 @@ declare namespace NPM {
         Conf: ConfigStatic;
         defs: ConfigDefs;
 
-        get(setting: string): string;
-        set(setting: string, value: string): void;
+        get<T>(setting: string): T;
```
[the remark this answers is not in the document]
Comment (styfle) [reply types/npm/index.d.ts]: Fixed

File: types/npm/index.d.ts [hunk 2]
```diff
@@ -167,8 +167,8 @@ declare namespace NPM {
         Conf: ConfigStatic;
         defs: ConfigDefs;
 
-        get(setting: string): string;
-        set(setting: string, value: string): void;
+        get(setting: string): any;
+        set<T>(setting: string, value: T): void;
```
Comment (SimonSchick) [inline types/npm/index.d.ts]: This is considered a `useless generic` as it doesn't contribute to type safety at all, please remove it.

**Review by SimonSchick — CHANGES_REQUESTED** [review]

File: types/npm/index.d.ts (hunk 2 above)
Comment (styfle) [reply types/npm/index.d.ts]: So should I change to `value: any` ?

Comment (SimonSchick) [reply types/npm/index.d.ts]: Yes.

Comment (styfle) [reply types/npm/index.d.ts]: Ok done

**Review by SimonSchick — APPROVED** [review]

**Review by amcasey — APPROVED** [review]

**Pull request merged**